flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/factories/DataGenTableSourceFactory.java

```diff
@@ -94,128 +97,176 @@ public String factoryIdentifier() {
 
 	@Override
 	public DynamicTableSource createDynamicTableSource(Context context) {
+		createTableFactoryHelper(this, context).validateExcept(FIELDS);
```

Comment:
Is this necessary, given that we'll call `FactoryUtil.validateFactoryOptions` and `FactoryUtil.validateUnconsumedKeys` later on?

Reply:
No necessary.